File: mobile/src/types/invite.ts
import { InviteStatus } from './common';

export interface InviteResponse {
  id: string;
  session_id: string;
  student_id: string;
  status: InviteStatus;
  session_title: string | null;
  session_type: string | null;
  session_mode: string | null;
  start_time: string | null;
  duration_minutes: number | null;
  tutor_name: string | null;
  location_city: string | null;
  conflict_details: Record<string, unknown> | null;
  created_at: string;
}

export interface InviteActionRequest {
  note?: string;
}

File: backend/app/schemas/invite.py
import uuid
from datetime import datetime

from pydantic import BaseModel, Field


class InviteResponse(BaseModel):
    id: uuid.UUID
    session_id: uuid.UUID
    student_id: uuid.UUID
    status: str
    session_title: str | None = None
    session_type: str | None = None
    session_mode: str | None = None
    start_time: datetime | None = None
    duration_minutes: int | None = None
    tutor_name: str | None = None
    location_city: str | None = None
    conflict_details: dict | None = None
    created_at: datetime

    model_config = {"from_attributes": True}


class InviteActionRequest(BaseModel):
    note: str | None = Field(None, max_length=500)
